protocol/communities/manager.go

```diff
@@ -4129,3 +4117,51 @@ func (m *Manager) fixupChannelMembers() error {
 
 	return nil
 }
+
+func (m *Manager) ReevaluatePrivelegedMember(community *Community, tokenPermissions []*protobuf.CommunityTokenPermission,
+	accountsAndChainIDs []*AccountChainIDsCombination, memberPubKey *ecdsa.PublicKey,
+	privilegedRole protobuf.CommunityMember_Roles, hasPrivilegedRole bool) (bool, error) {
+
+	hasPrivilegedRolePermissions := len(tokenPermissions) > 0
+	removeCurrentRole := false
+
+	if hasPrivilegedRolePermissions {
+		permissionResponse, err := m.checkPermissionToJoin(tokenPermissions, accountsAndChainIDs, true)
```

Comment:
That's misleading, AFAICT we are not checking permissions to join here.
```suggestion
		permissionResponse, err := m.checkPermissions(tokenPermissions, accountsAndChainIDs, true)
```